crates/assists/src/handlers/replace_string_with_char.rs

```diff
@@ -0,0 +1,127 @@
+use std::borrow::Cow;
+
+use syntax::{
+    ast::{self, HasQuotes, HasStringValue},
+    AstToken,
+    SyntaxKind::{RAW_STRING, STRING},
+    TextRange, TextSize,
+};
+use test_utils::mark;
+
+use crate::{AssistContext, AssistId, AssistKind, Assists};
+
+// Assist: replace_string_with_char
+//
+// Replace string with char
+//
+// ```
+// fn main() {
+//     find("{<|>");
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     find('{');
+// }
+// ```
+pub(crate) fn replace_string_with_char(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let token = ctx.find_token_at_offset(STRING).and_then(ast::String::cast)?;
+    let value = token.value()?;
+    let target = token.syntax().text_range();
+    if value.len() > 1 || value.is_empty() {
```

Comment:
This only works for ASCII characters, but other single characters might take up more than one byte.

Reply:
Thanks ! Updated :)